### PR TITLE
Add tests for docker resource limits

### PR DESCRIPTION
**Pull request by stefwalter**
I've split this out of #4600

### CODE REVIEW

**Comment by dperpeet** [comment]
tests are failing, sorry

File: pkg/docker/docker.css [hunk 1]
```diff
@@ -406,3 +406,10 @@
     border: 1px solid #d7d7d7;
     cursor: auto;
 }
+
+.size-text {
```
Comment (dperpeet) [inline pkg/docker/docker.css]: shouldn't we add `ct`? maybe `.size-ct-text`

File: pkg/docker/overview.js [hunk 2]
```diff
@@ -234,86 +234,56 @@ define([
...
-                    if (value && value[0] == "Data Space Total")
-                        total = value[1];
-                    else if (value && value[0] == "Data Space Used")
-                        used = value[1];
-                    else if (value && value[0] == "Data Space Available")
-                        avail = value[1];
-                });
+            var used;
+            var total;
+            var avail;
```
Comment (dperpeet) [inline pkg/docker/overview.js]: I know this is legacy code, but when I read `avail`, the first two things that pop into my mind are the noun and verb `avail`, not the abbreviation of `available`.

File: pkg/docker/util.js [hunk 3]
```diff
@@ -295,131 +295,157 @@ define([
...
-            if (slider.disabled) {
-                limit = undefined;
-                return _("unlimited");
-            }
-            limit = Math.round(slider.value * max);
-            if (limit < min)
-                limit = min;
-            return cockpit.format_bytes(limit, 1024);
+        /* Logarithmic scale */
+        var minv = Math.log(min);
```
Comment (dperpeet) [inline pkg/docker/util.js]: Should we do some checking here that `min` and `max` are `> 0`?

**Comment by dperpeet** [comment]
Setting <1024 cpu shares can lead to an error, do we need to catch this?
It has been fixed upstream in 1.11, so maybe not:
screenshot from 2016-06-23 09-24-04
 https://github.com/docker/docker/issues/22818

**Comment by dperpeet** [comment]
Changing the resources for a running container on my system didn't work. I opened the dialog to change, unchecked cpu and/or memory, but on closing the changes didn't apply.
```
Installed Packages
Name        : docker
Arch        : x86_64
Epoch       : 2
Version     : 1.10.3
Release     : 9.git667d6d1.fc24
```

File: pkg/docker/docker.css (hunk 1 above)
Comment (stefwalter) [inline pkg/docker/docker.css]: Done with: .size-text-ct

File: pkg/docker/overview.js (hunk 2 above)
Comment (stefwalter) [inline pkg/docker/overview.js]: Done.

File: pkg/docker/util.js (hunk 3 above)
Comment (stefwalter) [inline pkg/docker/util.js]: Added.

**Comment by stefwalter** [comment]
> Changing the resources for a running container on my system didn't work. I opened the dialog to change, unchecked cpu and/or memory, but on closing the changes didn't apply.

Cannot reproduce. On Fedora 23 I started a busybox container with limits on both CPU and Memory. I then went and unchecked both of those and they reset to unlimited and to defaults. 

Could you blow by blow outline a more clear reproducer using one of the test vms?

**Comment by dperpeet** [comment]
On the current f24 testvm:
 - create busybox with cpu/memory limits enabled, at default
 - run with /bin/sleep 100000000
 - go to container, change resource limits
 - uncheck both
 - open "change resource limits" dialog again, CPU priority will still be checked

**Comment by stefwalter** [comment]
Okay, hopefully this fixes the issue. No more checkbox in the Change dialog for CPU priority. It doesn't make sense there.

File: test/verify/check-docker [hunk 4]
```diff
@@ -343,6 +346,73 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+        b.set_val("#containers-run-image-name", "RESOURCE")
+        b.set_val("#containers-run-image-command", "/bin/sleep 1000000")
+        b.click("#containers-run-image-with-terminal");
+        if memory_supported:
+            b.click("#containers-run-image-memory input[type='checkbox']");
+            b.set_val("#containers-run-image-memory input.size-text", "236");
+        else:
+            b.wait_not_visible("#containers-run-image-memory")
+        b.click("#containers-run-image-cpu input[type='checkbox']");
+        b.set_val("#containers-run-image-cpu input.size-text", "512");
```
Comment (dperpeet) [inline test/verify/check-docker]: needs to be `size-text-ct`

File: test/verify/check-docker [hunk 5]
```diff
@@ -343,6 +346,73 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+
+        # Run one container without restarting
+        b.click('#containers-images tr:contains("busybox:latest") button.fa-play')
+        b.wait_popup("containers_run_image_dialog")
+        b.set_val("#containers-run-image-name", "RESOURCE")
+        b.set_val("#containers-run-image-command", "/bin/sleep 1000000")
+        b.click("#containers-run-image-with-terminal");
+        if memory_supported:
+            b.click("#containers-run-image-memory input[type='checkbox']");
+            b.set_val("#containers-run-image-memory input.size-text", "236");
```
Comment (dperpeet) [inline test/verify/check-docker]: needs to be `size-text-ct`

File: test/verify/check-docker [hunk 6]
```diff
@@ -343,6 +346,73 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+        b.wait_visible("#container-details")
+        if memory_supported:
+            b.wait_in_text("#container-details-memory-row", "/ 236 MiB")
+        b.wait_in_text("#container-details-cpu-row", "512 shares")
+
+        # Now adjust things
+        b.click("#container-details-resource-row button")
+        b.wait_popup("container-resources-dialog")
+        if memory_supported:
+            b.set_val(".memory-slider input.size-text", "246");
```
Comment (dperpeet) [inline test/verify/check-docker]: needs to be `size-text-ct`

File: test/verify/check-docker [hunk 7]
```diff
@@ -343,6 +346,73 @@ CMD ["/listen-on-port.sh", "%(port)d", "%(message)s"]
...
+        else:
+            b.wait_not_visible(".memory-slider")
+        b.set_val(".cpu-slider input.size-text", "256");
```
Comment (dperpeet) [inline test/verify/check-docker]: needs to be `size-text-ct`

**Comment by stefwalter** [comment]
Fixed the size-text-ct issues.

**Comment by dperpeet** [comment]
testResources fails on fedora-24

**Comment by stefwalter** [comment]
> testResources fails on fedora-24

I've seen this elsewhere. Holding off on fixing this until the run dialog is refactored.

**Pull request closed without merging**